ha/core/event_manager/event_manager.py

```diff
@@ -57,42 +89,181 @@ def _validate_component(self, component: str) -> None:
         if component.lower() not in SUBSCRIPTION_LIST:
             raise InvalidComponent(f"Invalid component {component}, not part of subscription list.")
 
-    def _validate_events(self, events: list) -> None:
-        """
-        Raise error InvalidEvent if event is not valid.
+    def subscribe(self, component : str, events : list):
+        '''
+           Register all the events for the notification. It maintains
+           list of events registered by the components. It maps the event
+           name with component name and store it in the consul. \
+           Also, maps the component with event
 
-        Args:
-            events (list): Event list.
-        """
-        if not isinstance(events, list):
-            raise InvalidEvent(f"Invalid type {events}, event type should be list")
-        for event in events:
-            if event not in EVENTS:
-                raise InvalidEvent(f"Invalid event: {event}, not part of HA event list.")
-        Log.debug(f"event: {event} is valid for subscription request")
-
-    def subscribe(self, component: str, events: list) -> str:
-        """
-        Register events for the notification. It maintains list of events registered by the components.
-
-        Args:
+           Args:
             component (str): Component Name
             events (list): Events.
-        """
-        pass
+        '''
+        Log.info(f'Received a subscribe request from {component} for event: {events}')
+        EventManager._validate_component(component)
+        EventManager._validate_events(events)
+        if component:
+            message_type = self._create_message_type(component)
+        resource_type = events.resource_type
```

Comment:
"events" here will be a list of SubscribeEvent objects, so we need to iterate over this list here.

Reply:
Done